libnetwork/internal/l2disco/unsol_na_linux.go

```diff
@@ -0,0 +1,80 @@
+package l2disco
+
+import (
+	"fmt"
+	"net"
+	"slices"
+
+	"golang.org/x/sys/unix"
+)
+
+var naTemplate = []byte{
+	0x88,       // Type (136=NA)
+	0x00,       // Code (always 0)
+	0x00, 0x00, // Checksum (filled in by the kernel)
+	0x20,             // Flags, Router=0, Solicited=0, Override=1
+	0x00, 0x00, 0x00, // Reserved
+	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, // Target IP
+	0x02,                               // Option - target link layer address
+	0x01,                               // Option length (32)
+	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, // Target MAC
+}
+
+type UnsolNA struct {
+	pkt []byte
+	sd  int
+	sa  *unix.SockaddrInet6
+}
+
+// NewUnsolNA returns a pointer to an object that can send unsolicited Neighbour
+// Advertisements on the interface with ifIndex, for ip and mac.
+// https://datatracker.ietf.org/doc/html/rfc4861#section-4.4
+func NewUnsolNA(ip net.IP, mac net.HardwareAddr, ifIndex int) (*UnsolNA, error) {
+	sd, err := unix.Socket(unix.AF_INET6, unix.SOCK_RAW, unix.IPPROTO_ICMPV6)
```

Comment:
We need to do the same here.

Reply:
Done (and the ones in the test code for good measure).